CONTRIBUTING.md

```diff
@@ -0,0 +1,138 @@
+install `sigstore` as an editable package into this environment.
+
+Any changes you make to the `sigstore` source tree will take effect
+immediately in the virtual environment.
+
+### Linting
+
+You can lint locally with:
+
+```bash
+make lint
+```
+
+`sigstore` is automatically linted and formatted with a collection of tools:
+
+* [`black`](https://github.com/psf/black): Code formatting
+* [`isort`](https://github.com/PyCQA/isort): Import sorting, ordering
+* [`flake8`](https://flake8.pycqa.org/en/latest/): PEP-8 linting, style enforcement
+* [`mypy`](https://mypy.readthedocs.io/en/stable/): Static type checking
+* [`interrogate`](https://interrogate.readthedocs.io/en/latest/): Documentation coverage
+
+
+### Testing
+
+You can run the tests locally with:
+
+```bash
+make test
+```
+
+You can also filter by a pattern (uses `pytest -k`):
+
+```bash
+make test TESTS=test_version
+```
+
+To test a specific file:
+
+```bash
+make test T=path/to/file.py
+```
+
+`sigstore` has a [`pytest`](https://docs.pytest.org/)-based unit test suite,
+including code coverage with [`coverage.py`](https://coverage.readthedocs.io/).
+
+### Documentation
+
+If you're running Python 3.7 or newer, you can run the documentation build locally:
+
+```bash
+make doc
+```
+
+`sigstore` uses [`pdoc3`](https://github.com/pdoc3/pdoc) to generate HTML documentation for
+the public Python APIs.
+
+Live documentation for the `main` branch is hosted
+[here](https://trailofbits.github.io/sigstore/). Only the public APIs are
+documented, all undocumented APIs are **intentionally private and unstable.**
+
+### Releasing
+
+**NOTE**: If you're a non-maintaining contributor, you don't need the steps
+here! They're documented for completeness and for onboarding future maintainers.
+
+Releases of `sigstore` are managed with [`bump`](https://github.com/di/bump)
+and GitHub Actions.
+
+```bash
+# default release (patch bump)
+make release
+
+# override the default
+# vX.Y.Z -> vX.Y.Z-rc.0
+make release BUMP_ARGS="--pre rc.0"
+
+# vX.Y.Z -> vN.0.0
+make release BUMP_ARGS="--major"
+```
+
+`make release` will fail if there are any untracked changes in the source tree.
+
+If `make release` succeeds, you'll see an output like this:
+
+```
+RUN ME MANUALLY: git push origin main && git push origin vX.Y.Z
+```
+
+Run that last command sequence to complete the release.
+
+## Development practices
+
+Here are some guidelines to follow if you're working on a new feature or changes to
+`sigstore`'s internal APIs:
+
+* *Keep the `sigstore` APIs as private as possible*. Nearly all of `sigstore`'s
+APIs should be private and treated as unstable and unsuitable for public use.
+If you're adding a new module to the source tree, prefix the filename with an underscore to
+emphasize that it's an internal (e.g., `sigstore/_foo.py` instead of `sigstore/foo.py`).
+
+* *Perform judicious debug logging.* `sigstore` uses the standard Python
```

Comment:
This isn't the case for `sigstore-python` yet. Although I'm thinking we'll probably end up adding debug logging further down the track so it might not be worth removing this bit.

Reply:
Yeah, we should keep that in as an aspirational component of a stable release.